addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -3118,7 +3118,18 @@ const SnippetOptionWidget = Widget.extend({
      */
     selectDataAttribute: function (previewMode, widgetValue, params) {
         const value = this._selectAttributeHelper(widgetValue, params);
-        this.$target[0].dataset[params.attributeName] = value;
+        console.log(value)
+        if (params.attributeName === "max_files_number" && value > 1) {
+            this.$target[0].setAttribute("multiple", "multiple"); // add the "multiple" attribute to allow selecting multiple files
+            this.$target[0].dataset[params.attributeName] = value;
+        }
+        else if ((params.attributeName === "max_files_number" || params.attributeName === "max_file_size") && (value == 0 || value === null)) {
```

Comment:
You will learn that we are convention maniac :) So here are some conventions we want to follow: https://github.com/odoo/odoo/wiki/Javascript-coding-guidelines and especially with a sample linter configuration.

Here the lines I am commenting does not use the if..else form we use by convention :)